src/garrysmod/gamemodes/zombiesurvival/entities/weapons/weapon_zs_hammer/shared.lua

```diff
@@ -62,3 +64,55 @@ end
 function SWEP:IsOwnerBarricadeGhosting()
 	return false
 end
+
+-- Reloading at a door will attack it for vast damage
+function SWEP:ReloadAttack(target)
+	if not self:CanPrimaryAttack() then return end
+	self:SetNextAttack()
+	
+	local owner = self:GetOwner()
+
+	self:DoMeleeAttackAnim()
+
+	if not target.Hit then
+		if self.MissAnim then
+			self:SendWeaponAnim(self.MissAnim)
+		end
+		self.IdleAnimation = CurTime() + self:SequenceDuration()
+		self:PlaySwingSound()
+
+		if owner.MeleePowerAttackMul and owner.MeleePowerAttackMul > 1 then
+			self:SetPowerCombo(0)
+		end
+
+		if self.PostOnMeleeMiss then self:PostOnMeleeMiss(target) end
+
+		return
+	end
+
+	local hitent = target.Entity
+	
+	if self.HitAnim then
+		self:SendWeaponAnim(self.HitAnim)
+	end
+	self.IdleAnimation = CurTime() + self:SequenceDuration()
+
+	owner:EmitSound("weapons/melee/crowbar/crowbar_hit-"..math.random(4)..".ogg")
+	self:PlayHitSound()
+
+	if self.OnMeleeHit and self:OnMeleeHit(hitent, false, target) then
+		return
+	end
+
+	if SERVER then
+		self:ServerMeleeHitEntity(target, hitent, self.DoorDamageMultiplier)
+	end
+
+	self:MeleeHitEntity(target, hitent, self.DoorDamageMultiplier)
+
+	if self.PostOnMeleeHit then self:PostOnMeleeHit(hitent, hitflesh, target) end
+
+	if SERVER then
+		self:ServerMeleePostHitEntity(target, hitent, self.DoorDamageMultiplier)
+	end
+end
```

Comment:
I decided not to add more cruft into the base melee functions as this isn't really an attack (it just looks and acts similar).